app/controllers/flyover_comments/votes_controller.rb

```diff
@@ -0,0 +1,67 @@
+require_dependency "flyover_comments/application_controller"
+require_dependency "flyover_comments/authorization"
+
+module FlyoverComments
+  class VotesController < ApplicationController
+    include FlyoverComments::Authorization
+
+    respond_to :json, only: [:create, :destroy]
+
+    before_action :load_vote, only: [:update, :destroy]
+
+    def create
+      @comment = FlyoverComments::Comment.find(params[:comment_id])
+      authorize_flyover_vote_create!
+      @vote = @comment.votes.new(value: params[:value])
+      @vote._user = send(FlyoverComments.current_user_method.to_sym)
+
+      if @vote.save
+        @comment.recalculate_vote_counts!
```

Comment:
We can (and should) put this in the model, I think. ActiveRecord::Dirty will let you see if the value of a field changed.

So in comment.rb we'd have something like:
```
before_create :increment_comment_counter_cache
before_update :update_comment_counter_caches_for_value_change
after_destroy :decrement_comment_counter_cache

def increment_comment_counter_cache
  FlyoverComments::Comment.increment_counter(:upvotes_count, comment_id) if upvote?
  FlyoverComments::Comment.increment_counter(:downvotes_count, comment_id) if downvote?
end

def decrement_comment_counter_cache
  FlyoverComments::Comment.decrement_counter(:upvotes_count, comment_id) if upvote?
  FlyoverComments::Comment.decrement_counter(:downvotes_count, comment_id) if downvote?
end

def update_comment_counter_caches_for_value_change
  if value_changed?
    if upvote?
      FlyoverComments::Comment.increment_counter(:upvotes_count, comment_id)
      FlyoverComments::Comment.decrement_counter(:downvotes_count, comment_id)
    else
      FlyoverComments::Comment.increment_counter(:downvotes_count, comment_id)
      FlyoverComments::Comment.decrement_counter(:upvotes_count, comment_id)
    end
  end
end

```